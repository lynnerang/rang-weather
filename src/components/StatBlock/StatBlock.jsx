import React, { Component } from 'react';
import { Line } from 'react-chartjs-2';
import PropTypes from 'prop-types';

export class StatBlock extends Component {
	render() {
		const { hasChart, hasNums, type, title, stats } = this.props || '';
		const { stat1, stat2, unit, label1, label2 } = hasNums ? stats || '' : '';
		const unit2 = stat2 ? unit : '';
		const chartHeight = hasNums && hasChart ? 60 : 70;
		const isShort = hasNums && hasChart ? '' : 'block-short';

		const metrics =
			hasNums && hasChart
				? this.props.data.map(i => {
						const result = i.stat2 && type !== 'wind' ? parseInt(i.stat2) : parseInt(i.stat1);
						return result;
					})
				: this.props.data;

		const chartData = {
			labels: this.props.timestamps,
			datasets: [
				{
					maintainAspectRatio: false,
					borderColor: 'white',
					data: metrics
				}
			]
		};

		const options = {
			legend: {
				display: false
			},
			elements: {
				line: {
					tension: 0
				},
				point: {
					radius: 0
				}
			},
			scales: {
				yAxes: [
					{
						ticks: {
							fontSize: 10,
							fontColor: 'rgba(0, 0, 0, 0.5)',
							color: 'rgba(255, 255, 255, 0.3)'
						},
						gridLines: {
							color: 'rgba(255, 255, 255, 0.3)',
							drawBorder: false
						}
					}
				],
				xAxes: [
					{
						ticks: {
							fontSize: 10,
							fontColor: 'rgba(0, 0, 0, 0.5)',
							color: 'rgba(255, 255, 255, 0.3)'
						},
						gridLines: {
							color: 'rgba(255, 255, 255, 0.3)'
						}
					}
				]
			}
		};

		const chart = hasChart ? (
			<div className="StatBlock-graph">
				<Line data={chartData} height={chartHeight} options={options} />
			</div>
		) : null;

		return (
			<article className={`StatBlock ${type} ${isShort}`}>
				<img className="StatBlock-bg" src={require(`../../images/${type}bg.png`)} alt="" />
				<div className="StatBlock-header">
					<img className="StatBlock-icon" src={require(`../../images/${type}icon.png`)} alt={`${type} icon`} />
					<h3>{title}</h3>
				</div>
				<div className="StatBlock-metrics">
					<div className="metric left-metric">
						<p className="metric-number">
							{stat1}
							<span className="unit-txt">{unit}</span>
						</p>
						<p className={`metric-label metric-label-${type}`}>{label1}</p>
					</div>
					<div className="metric right-metric">
						<p className="metric-number">
							{stat2}
							<span className="unit-txt">{unit2}</span>
						</p>
						<p className={`metric-label metric-label-${type}`}>{label2}</p>
					</div>
				</div>
				{chart}
			</article>
		);
	}
}

StatBlock.propTypes = {
	type: PropTypes.string,
	title: PropTypes.string,
	stats: PropTypes.object,
	data: PropTypes.array,
	timestamps: PropTypes.array,
	hasChart: PropTypes.bool,
	hasNums: PropTypes.bool
};

export default StatBlock;
