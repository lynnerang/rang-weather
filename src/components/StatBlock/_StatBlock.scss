.StatBlock {
  width: 100vw;
  height: 165px;
  position: relative;

  &.block-short {
    height: 120px;

    .StatBlock-graph {
      margin-top: 2px;
    }
  }

  .StatBlock-bg {
    height: 100%;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
  }

  .StatBlock-header {
    padding: 10px;
    display: flex;

    .StatBlock-icon {
      margin-top: 2px;
      margin-right: 3px;
      height: 15px;
    }

    h3 {
      text-transform: uppercase;
      text-align: left;
      font-size: 15px;
      font-weight: 400;
      margin: 0;
    }
  }

  .StatBlock-metrics {
    display: flex;
    justify-content: space-around;

    .metric {
      display: flex;
      flex-direction: column;

      .metric-number {
        margin-top: -8px;
        margin-bottom: -13px;
        font-size: 54px;
        font-family: 'PT Sans Narrow', sans-serif;

        span {
          font-size: 16px;
        }
      }
  
      .metric-label {
        font-size: 13px;
        margin: 0;
        font-weight: 500;
        min-height: 15px;
      }

      .metric-label-temp {
        color: #CE5E02;
      }

      .metric-label-humidity {
        color: #4479A0;
      }

      .metric-label-wind {
        color: #44644B;
      }

      .metric-label-temphilo {
        color: rgb(60, 59, 108);
      }

      .metric-label-sun {
        color: rgb(136, 58, 24);
      }

      .metric-label-storm {
        color: rgb(198, 197, 197);
      }
    }
  }

  .StatBlock-graph {
    width: 100%;
  }

}

@import url('https://fonts.googleapis.com/css?family=Wallpoet');

.compass {
  position: absolute;
  top: 22px;
  right: 21px;
	width: 120px;
	height: 120px;
	overflow: hidden;
}

.compass .disc {
  position: absolute;
  right: 20px;
	z-index: 200;
	background: url('https://i.ibb.co/7jR95bx/compass2.png');
	width: 65px;
	height: 65px;
	background-size: 100%;
}

.compass .arrow {
	position: absolute;
  right: 20px;
	z-index: 1000;
	background: url('http://dev.rvltn.eu/compass/arrow.svg');
	width: 65px;
	height: 65px;
	background-size: 100%;
  transform: rotate(var(--degrees));
  transition: 0.75s all;
}