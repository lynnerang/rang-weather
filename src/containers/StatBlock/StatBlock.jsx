import React, { Component } from 'react';

class StatBlock extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { type, title } = this.props;
    const { stat1, stat2, unit, label1, label2 } = this.props.stats || '';
    const unit2 = stat2 ? unit : '';

    return (
      <article className={`StatBlock ${type}`}>
        <img className="StatBlock-bg" src={require(`../../images/${type}bg.png`)} />
        <div className="StatBlock-header">
          <img className="StatBlock-icon" src={require(`../../images/${type}icon.png`)} alt={`${type} icon`}/>
          <h2>{title}</h2>
        </div>
        <div className="StatBlock-metrics">
          <div className="metric left-metric">
            <p className="metric-number">{stat1}<span className="unit-txt">{unit}</span></p>
            <p className={`metric-label metric-label-${type}`}>{label1}</p>
          </div>
          <div className="metric right-metric">
            <p className="metric-number">{stat2}<span className="unit-txt">{unit2}</span></p>
            <p className={`metric-label metric-label-${type}`}>{label2}</p>
          </div>
        </div>
        <div className="StatBlock-graph">
          
        </div>
      </article>
    );
  }
}

export default StatBlock;