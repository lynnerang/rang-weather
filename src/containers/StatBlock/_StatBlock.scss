.StatBlock {
  width: 100vw;
  height: 165px;
  position: relative;

  .StatBlock-bg {
    height: 100%;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
  }

  .StatBlock-header {
    padding: 10px;
    display: flex;

    .StatBlock-icon {
      margin-top: 2px;
      margin-right: 3px;
      height: 15px;
    }

    h2 {
      text-transform: uppercase;
      text-align: left;
      font-size: 15px;
      font-weight: 400;
      margin: 0;
    }
  }

  .StatBlock-metrics {
    display: flex;
    justify-content: space-around;

    .metric {
      display: flex;
      flex-direction: column;

      .metric-number {
        margin-top: -8px;
        margin-bottom: -12px;
        font-size: 54px;
        font-family: 'PT Sans Narrow', sans-serif;

        span {
          font-size: 16px;
        }
      }
  
      .metric-label {
        font-size: 13px;
        margin: 0;
        font-weight: 500;

      }

      .metric-label-temp {
        color: #CE5E02;
      }

      .metric-label-humidity {
        color: #4479A0;
      }

      .metric-label-wind {
        color: #44644B;
      }
    }
  }

  .StatBlock-graph {
    width: 100%;
    margin-top: 12px;
    height: 40px;
    position: absolute;
    bottom: 20px;
  }

}