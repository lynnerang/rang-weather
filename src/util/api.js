import { baseUrl, apiKey, appKey, macAddress, darkskyApi } from '../api/utilities';

export const fetchCurrentStats = async () => {
  const res = await fetch(`${baseUrl}?applicationKey=${appKey}&apiKey=${apiKey}`)
  if (!res.ok) {
    throw new Error('Failed to fetch current stats data.');
  }
  return await res.json();
};

export const fetchHistoricalStats = async () => {
  const res = await fetch(`${baseUrl}/${macAddress}?applicationKey=${appKey}&apiKey=${apiKey}`)
  if (!res.ok) {
    throw new Error('Failed to fetch historical stats data.');
  }
  return await res.json();
};

export const fetchForecastData = async () => {
  const res = await fetch(`https://cors-anywhere.herokuapp.com/https://api.darksky.net/forecast/${darkskyApi}/40.0114669,-105.0940768?exclude=minutely,alerts,flags`)
  if (!res.ok) {
    throw new Error('Failed to fetch forecast data.');
  }
  return await res.json();
};

