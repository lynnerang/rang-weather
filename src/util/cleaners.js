export const cleanStats = stats => {
  return ({
      timestamps: stats.date.map(i => parseInt(i.date.split(':')[0].split('T')[1], 10)),
      temp: {
        stat1: stats.tempinf,
        stat2: stats.tempf,
        unit: '°F',
        label1: 'Indoor',
        label2: 'Outdoor'
      },
      humidity: {
        stat1: (stats.humidityin*100).toFixed(1),
        stat2: (stats.humidity*100).toFixed(1),
        unit: '%',
        label1: 'Indoor',
        label2: 'Outdoor'
      },
      pressure: {
        stat1: stats.baromrelin,
        unit: 'REL.lnhg'
      },
      dew: {
        stat1: stats.dewPoint,
        unit: '°F'
      },
      wind: {
        stat1: (stats.windspeedmph * .447).toFixed(2),
        stat2: stats.winddir,
        unit: 'M/SEC',
        label1: 'Speed',
        label2: 'Direction'
      }
    })
} 

export const cleanHistoricalStats = snapshots => {
  const hourly = snapshots.filter(instance => {
    const mins = instance.date.split(':')[1];
    if (mins === '00') {
      return instance;
    }
  });
  const last12s = hourly.slice(0, 12).reverse();

  return last12s.map(record => cleanStats(record));
}

export const cleanForecastData = data => {
  const next12HrData = data.hourly.data.slice(1, 13);
  const next12HrTimes = next12HrData.map(hour => `${new Date(hour.time * 1000).getHours()}:`);
  const sunriseTime = new Date(data.daily.data[0].sunriseTime * 1000);
  const sunsetTime = new Date(data.daily.data[0].sunsetTime * 1000);
  const moonPhase = data.daily.data[0].moonPhase;
  let moonPhaseTxt;

  if (moonPhase === 0) {
    moonPhaseTxt = 'New Moon'
  } else if (moonPhase === '.25') {
    moonPhaseTxt = 'Quarter Moon'
  } else if (moonPhase === '.5') {
    moonPhaseTxt = 'Half Moon'
  } else if (moonPhase === '1') {
    moonPhaseTxt = 'Full Moon'
  } else {
    moonPhaseTxt = ''
  }

  return ({
    summary: data.hourly.summary,
    icon: data.hourly.icon,
    timestamps: next12HrTimes,
    temp: {
      stat1: data.daily.data[0].temperatureHigh.toFixed(1),
      stat2: data.daily.data[0].temperatureLow.toFixed(1),
      unit: '°F',
      label1: 'High',
      label2: 'Low'
    },
    storm: {
      stat1: data.currently.nearestStormDistance,
      stat2: data.currently.nearestStormBearing,
      unit: 'miles',
      label1: 'Distance',
      label2: 'Direction'
    },
    sun: {
      stat1: `${sunriseTime.getHours()}:${sunriseTime.getMinutes()}`,
      label1: 'Sunrise',
      stat2: `${sunsetTime.getHours()}:${sunsetTime.getMinutes()}`,
      label2: 'Sunset'
    },
    moon: {
      stat1: Math.floor(moonPhase * 100),
      unit: '% FULL',
      label1: moonPhaseTxt
    },
    tempTrend: next12HrData.map(hour => hour.temperature.toFixed(1)),
    humidityTrend: next12HrData.map(hour => (hour.humidity*100).toFixed(1)),
    pressureTrend: next12HrData.map(hour => hour.pressure),
    dewTrend: next12HrData.map(hour => hour.dewPoint),
    windTrend: next12HrData.map(hour => hour.windSpeed)
  })
}

