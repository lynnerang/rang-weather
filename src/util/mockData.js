export const mockRawCurrentStats = {
  dateutc: 1559334720000,
  winddir: 94,
  windspeedmph: 1.8,
  windgustmph: 2.2,
  maxdailygust: 10.3,
  tempf: 75.2,
  hourlyrainin: 0,
  eventrainin: 0,
  dailyrainin: 0,
  weeklyrainin: 0.98,
  monthlyrainin: 0.98,
  totalrainin: 3.88,
  baromrelin: 30.17,
  baromabsin: 24.82,
  humidity: 24,
  tempinf: 75.7,
  humidityin: 33,
  uv: 8,
  solarradiation: 829.44,
  feelsLike: 73.55,
  dewPoint: 35.92,
  date: '2019-05-31T20:32:00.000Z'
};

export const mockCurrentStats = {
	dew: {
    stat1: 35.92,
		unit: '°F'
	},
	humidity: {
    stat1: 33,
    stat2: 24,
		unit: '%',
		label1: 'Indoor',
		label2: 'Outdoor'
	},
	pressure: {
		stat1: 30.17,
		unit: 'REL.lnhg'
	},
	temp: {
		stat1: 75.7,
		stat2: 75.2,
		unit: '°F',
		label1: 'Indoor',
		label2: 'Outdoor'
	},
	wind: {
		stat1: .8,
		stat2: 94,
		unit: 'M/SEC',
		label1: 'Speed',
		label2: 'Direction'
	}
};

export const mockHistoricalStats = [
	{
		dew: {
			stat1: 47.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 36,
			stat2: 41,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 30.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 77.7,
			stat2: 72.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:02:00.000Z'
		},
		wind: {
			stat1: 1.1175,
			stat2: 3,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:04:00.000Z'
		}
	},
	{
		dew: {
			stat1: 37.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 46,
			stat2: 31,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 32.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 71.7,
			stat2: 70.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		wind: {
			stat1: 2.1175,
			stat2: 4.2,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	},
	{
		dew: {
			stat1: 47.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 36,
			stat2: 41,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 30.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 77.7,
			stat2: 72.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:03:00.000Z'
		},
		wind: {
			stat1: 1.12,
			stat2: 3,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	},
	{
		dew: {
			stat1: 37.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 46,
			stat2: 31,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 32.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 71.7,
			stat2: 70.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		wind: {
			stat1: 2.11,
			stat2: 4.2,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	},
	{
		dew: {
			stat1: 47.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 36,
			stat2: 41,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 30.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 77.7,
			stat2: 72.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		wind: {
			stat1: 1.11,
			stat2: 3,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	},
	{
		dew: {
			stat1: 37.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 46,
			stat2: 31,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 32.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 71.7,
			stat2: 70.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		wind: {
			stat1: 2.11,
			stat2: 4.2,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	},
	{
		dew: {
			stat1: 47.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 36,
			stat2: 41,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 30.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 77.7,
			stat2: 72.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		wind: {
			stat1: 1.11,
			stat2: 3,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	},
	{
		dew: {
			stat1: 37.63,
			unit: '°F',
			date: '2019-06-01T20:30:00.000Z'
		},
		humidity: {
			stat1: 46,
			stat2: 31,
			unit: '%',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		pressure: {
			stat1: 32.15,
			unit: 'REL.lnhg',
			date: '2019-06-01T20:30:00.000Z'
		},
		temp: {
			stat1: 71.7,
			stat2: 70.7,
			unit: '°F',
			label1: 'Indoor',
			label2: 'Outdoor',
			date: '2019-06-01T20:30:00.000Z'
		},
		wind: {
			stat1: 2.11,
			stat2: 4.2,
			unit: 'M/SEC',
			label1: 'Speed',
			label2: 'Direction',
			date: '2019-06-01T20:30:00.000Z'
		}
	}
];
