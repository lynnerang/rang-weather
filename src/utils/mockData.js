export const mockCurrentStats = {
    lastData: {
    dateutc: 1559334720000,
    winddir: 94,
    windspeedmph: 1.8,
    windgustmph: 2.2,
    maxdailygust: 10.3,
    tempf: 75.2,
    hourlyrainin: 0,
    eventrainin: 0,
    dailyrainin: 0,
    weeklyrainin: 0.98,
    monthlyrainin: 0.98,
    totalrainin: 3.88,
    baromrelin: 30.17,
    baromabsin: 24.82,
    humidity: 24,
    tempinf: 75.7,
    humidityin: 33,
    uv: 8,
    solarradiation: 829.44,
    feelsLike: 73.55,
    dewPoint: 35.92,
    date: '2019-05-31T20:32:00.000Z'
  }
}

export const mockHistoricalStats = [
  {
    dateutc: 1559421600000,
    winddir: 171,
    windspeedmph: 0.2,
    windgustmph: 1.1,
    maxdailygust: 12.5,
    tempf: 70.9,
    hourlyrainin: 0,
    eventrainin: 0,
    dailyrainin: 0,
    weeklyrainin: 0.98,
    monthlyrainin: 0,
    totalrainin: 3.88,
    baromrelin: 30.19,
    baromabsin: 24.85,
    humidity: 44,
    tempinf: 75.2,
    humidityin: 37,
    uv: 4,
    solarradiation: 489.34,
    feelsLike: 69.76,
    dewPoint: 47.89,
    date: '2019-06-01T20:40:00.000Z'
  },
  {
    dateutc: 1559421300000,
    winddir: 355,
    windspeedmph: 3.4,
    windgustmph: 3.4,
    maxdailygust: 12.5,
    tempf: 69.6,
    hourlyrainin: 0,
    eventrainin: 0,
    dailyrainin: 0,
    weeklyrainin: 0.98,
    monthlyrainin: 0,
    totalrainin: 3.88,
    baromrelin: 30.19,
    baromabsin: 24.85,
    humidity: 47,
    tempinf: 75.4,
    humidityin: 37,
    uv: 5,
    solarradiation: 537.06,
    feelsLike: 68.47,
    dewPoint: 48.47,
    date: '2019-06-01T20:35:00.000Z'
  },
  {
    dateutc: 1559421000000,
    winddir: 1,
    windspeedmph: 1.3,
    windgustmph: 3.4,
    maxdailygust: 12.5,
    tempf: 69.4,
    hourlyrainin: 0,
    eventrainin: 0,
    dailyrainin: 0,
    weeklyrainin: 0.98,
    monthlyrainin: 0,
    totalrainin: 3.88,
    baromrelin: 30.19,
    baromabsin: 24.85,
    humidity: 49,
    tempinf: 75.2,
    humidityin: 37,
    uv: 3,
    solarradiation: 382.18,
    feelsLike: 68.34,
    dewPoint: 49.4,
    date: '2019-06-01T20:30:00.000Z'
  }
]