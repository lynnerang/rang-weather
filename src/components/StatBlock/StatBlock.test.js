import React from 'react';
import { shallow } from 'enzyme';
import { StatBlock } from './StatBlock';
import { mockCurrentStats, mockHistoricalStats, mockForecastData } from '../../util/mockData';

describe('StatBlock', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<StatBlock
      type="wind"
      title="wind"
      stats={mockCurrentStats.wind}
      data={mockHistoricalStats.map(stat => stat.wind)}
      hasChart={true}
      hasNums={true}
    />)
  })

  it('matches the snapshot when it has both a chart and stats', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when it has stats but no chart', () => {
    wrapper = shallow(<StatBlock
      type="storm"
      title="storm risk"
      stats={mockCurrentStats.storm}
      hasChart={false}
      hasNums={true}
    />)

    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when it has a chart but no stats', () => {
    wrapper = shallow(<StatBlock
      type="temp"
      title="temperature"
      data={mockForecastData.tempTrend}
      hasChart={true}
      hasNums={false}
    />)

    expect(wrapper).toMatchSnapshot();
  })

});
