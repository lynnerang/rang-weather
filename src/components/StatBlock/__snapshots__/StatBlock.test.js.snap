// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatBlock matches the snapshot when it has both a chart and stats 1`] = `
<article
  className="StatBlock wind "
>
  <img
    alt=""
    className="StatBlock-bg"
    src="windbg.png"
  />
  <div
    className="StatBlock-header"
  >
    <img
      alt="wind icon"
      className="StatBlock-icon"
      src="windicon.png"
    />
    <h3>
      wind
    </h3>
  </div>
  <div
    className="StatBlock-metrics"
  >
    <div
      className="metric left-metric"
    >
      <p
        className="metric-number"
      >
        0.8
        <span
          className="unit-txt"
        >
          M/SEC
        </span>
      </p>
      <p
        className="metric-label metric-label-wind"
      >
        Speed
      </p>
    </div>
    <div
      className="metric right-metric"
    >
      <p
        className="metric-number"
      >
        94
        <span
          className="unit-txt"
        >
          M/SEC
        </span>
      </p>
      <p
        className="metric-label metric-label-wind"
      >
        Direction
      </p>
    </div>
  </div>
  <div
    className="StatBlock-graph"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "borderColor": "white",
              "data": Array [
                5,
              ],
              "maintainAspectRatio": false,
            },
          ],
          "labels": undefined,
        }
      }
      height={60}
      options={
        Object {
          "elements": Object {
            "line": Object {
              "tension": 0,
            },
            "point": Object {
              "radius": 0,
            },
          },
          "legend": Object {
            "display": false,
          },
          "scales": Object {
            "xAxes": Array [
              Object {
                "gridLines": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                },
                "ticks": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                  "fontColor": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 10,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "gridLines": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                  "drawBorder": false,
                },
                "ticks": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                  "fontColor": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 10,
                },
              },
            ],
          },
        }
      }
    />
  </div>
</article>
`;

exports[`StatBlock should match the snapshot when it has a chart but no stats 1`] = `
<article
  className="StatBlock temp block-short"
>
  <img
    alt=""
    className="StatBlock-bg"
    src="tempbg.png"
  />
  <div
    className="StatBlock-header"
  >
    <img
      alt="temp icon"
      className="StatBlock-icon"
      src="tempicon.png"
    />
    <h3>
      temperature
    </h3>
  </div>
  <div
    className="StatBlock-metrics"
  >
    <div
      className="metric left-metric"
    >
      <p
        className="metric-number"
      >
        <span
          className="unit-txt"
        />
      </p>
      <p
        className="metric-label metric-label-temp"
      />
    </div>
    <div
      className="metric right-metric"
    >
      <p
        className="metric-number"
      >
        <span
          className="unit-txt"
        />
      </p>
      <p
        className="metric-label metric-label-temp"
      />
    </div>
  </div>
  <div
    className="StatBlock-graph"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "borderColor": "white",
              "data": Array [
                "63.5",
                "60.8",
                "59.2",
                "58.5",
                "57.5",
                "55.3",
                "53.3",
                "51.6",
                "51.5",
                "56.1",
              ],
              "maintainAspectRatio": false,
            },
          ],
          "labels": undefined,
        }
      }
      height={70}
      options={
        Object {
          "elements": Object {
            "line": Object {
              "tension": 0,
            },
            "point": Object {
              "radius": 0,
            },
          },
          "legend": Object {
            "display": false,
          },
          "scales": Object {
            "xAxes": Array [
              Object {
                "gridLines": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                },
                "ticks": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                  "fontColor": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 10,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "gridLines": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                  "drawBorder": false,
                },
                "ticks": Object {
                  "color": "rgba(255, 255, 255, 0.3)",
                  "fontColor": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 10,
                },
              },
            ],
          },
        }
      }
    />
  </div>
</article>
`;

exports[`StatBlock should match the snapshot when it has stats but no chart 1`] = `
<article
  className="StatBlock storm block-short"
>
  <img
    alt=""
    className="StatBlock-bg"
    src="stormbg.png"
  />
  <div
    className="StatBlock-header"
  >
    <img
      alt="storm icon"
      className="StatBlock-icon"
      src="stormicon.png"
    />
    <h3>
      storm risk
    </h3>
  </div>
  <div
    className="StatBlock-metrics"
  >
    <div
      className="metric left-metric"
    >
      <p
        className="metric-number"
      >
        <span
          className="unit-txt"
        />
      </p>
      <p
        className="metric-label metric-label-storm"
      />
    </div>
    <div
      className="metric right-metric"
    >
      <p
        className="metric-number"
      >
        <span
          className="unit-txt"
        />
      </p>
      <p
        className="metric-label metric-label-storm"
      />
    </div>
  </div>
</article>
`;
